mlir/lib/Dialect/GPU/Transforms/SerializeToHsaco.cpp

```diff
@@ -410,6 +410,8 @@ SerializeToHsacoPass::createHsaco(const SmallVectorImpl<char> &isaBinary) {
     return {};
   }
   llvm::FileRemover cleanupHsaco(tempHsacoFilename);
+  // Prevent file descriptor leak.
+  llvm::sys::fs::closeFile(tempHsacoFD);
```

Comment:
```suggestion
  llvm::sys::fs::closeFile(llvm::sys::fs::convertFDToNativeFile(tempHsacoFD));
```

Reply:
There is a `createTemporaryFile` that does not take a `fd`, why not use this one instead?

Reply:
I didn't know that